### ISSUE TITLE
LinkedList

### ISSUE THREAD

**Issue by Lo0kadistraction** [JonnyBooker/split-flap#1]
ESPMaster will not compile due to error with LinkedList I have installed the linked list from the library manger. https://github.com/ivanseidel/LinkedList That did not seem to help. Please advise. Error message to follow

Arduino: 1.8.18 (Windows 10), Board: "Generic ESP8266 Module, 80 MHz, Flash, Disabled (new aborts on oom), Disabled, All SSL ciphers (most compatible), 32KB cache + 32KB IRAM (balanced), Use pgm_read macros for IRAM/PROGMEM, dtr (aka nodemcu), 26 MHz, 40MHz, DOUT (compatible), 1MB (FS:64KB OTA:~470KB), 2, nonos-sdk 2.2.1+100 (190703), v2 Lower Memory, Disabled, None, Only Sketch, 115200"

In file included from Z:\Projects\3D_Printer\Split_Flap\split-flap-master\ESPMaster\ESPMaster.ino:15:

C:\Users\guita\OneDrive\Documents\Arduino\libraries\LinkedList/LinkedList.h:26:7: error: redeclared with 1 template parameter

   26 | class LinkedList{

      |       ^~~~~~~~~~

In file included from C:\Users\guita\OneDrive\Documents\Arduino\libraries\ESPAsyncWebServer-master\src/ESPAsyncWebServer.h:29,

                 from Z:\Projects\3D_Printer\Split_Flap\split-flap-master\ESPMaster\ESPMaster.ino:10:

C:\Users\guita\OneDrive\Documents\Arduino\libraries\ESPAsyncWebServer-master\src/StringArray.h:39:7: note: previous declaration 'template<class T, template<class> class Item> class LinkedList' used 2 template parameters

   39 | class LinkedList {

      |       ^~~~~~~~~~

In file included from Z:\Projects\3D_Printer\Split_Flap\split-flap-master\ESPMaster\ESPMaster.ino:15:

C:\Users\guita\OneDrive\Documents\Arduino\libraries\LinkedList/LinkedList.h:113:27: error: invalid use of incomplete type 'class LinkedList<T>'

  113 | LinkedList<T>::LinkedList()

      |                           ^

In file included from C:\Users\guita\OneDrive\Documents\Arduino\libraries\ESPAsyncWebServer-master\src/ESPAsyncWebServer.h:29,

                 from Z:\Projects\3D_Printer\Split_Flap\split-flap-master\ESPMaster\ESPMaster.ino:10:

C:\Users\guita\OneDrive\Documents\Arduino\libraries\ESPAsyncWebServer-master\src/StringArray.h:39:7: note: declaration of 'class LinkedList<T>'

   39 | class LinkedList {

      |       ^~~~~~~~~~

In file included from Z:\Projects\3D_Printer\Split_Flap\split-flap-master\ESPMaster\ESPMaster.ino:15:

C:\Users\guita\OneDrive\Documents\Arduino\libraries\LinkedList/LinkedList.h:126:28: error: invalid use of incomplete type 'class LinkedList<T>'

  126 | LinkedList<T>::~LinkedList()

      |                            ^

In file included from C:\Users\guita\OneDrive\Documents\Arduino\libraries\ESPAsyncWebServer-master\src/ESPAsyncWebServer.h:29,

                 from Z:\Projects\3D_Printer\Split_Flap\split-flap-master\ESPMaster\ESPMaster.ino:10:

C:\Users\guita\OneDrive\Documents\Arduino\libraries\ESPAsyncWebServer-master\src/StringArray.h:39:7: note: declaration of 'class LinkedList<T>'

   39 | class LinkedList {

      |       ^~~~~~~~~~

In file included from Z:\Projects\3D_Printer\Split_Flap\split-flap-master\ESPMaster\ESPMaster.ino:15:

C:\Users\guita\OneDrive\Documents\Arduino\libraries\LinkedList/LinkedList.h:145:46: error: invalid use of incomplete type 'class LinkedList<T>'

  145 | ListNode<T>* LinkedList<T>::getNode(int index){

      |                                              ^

In file included from C:\Users\guita\OneDrive\Documents\Arduino\libraries\ESPAsyncWebServer-master\src/ESPAsyncWebServer.h:29,

                 from Z:\Projects\3D_Printer\Split_Flap\split-flap-master\ESPMaster\ESPMaster.ino:10:

C:\Users\guita\OneDrive\Documents\Arduino\libraries\ESPAsyncWebServer-master\src/StringArray.h:39:7: note: declaration of 'class LinkedList<T>'

   39 | class LinkedList {

      |       ^~~~~~~~~~

In file included from Z:\Projects\3D_Printer\Split_Flap\split-flap-master\ESPMaster\ESPMaster.ino:15:

C:\Users\guita\OneDrive\Documents\Arduino\libraries\LinkedList/LinkedList.h:176:25: error: invalid use of incomplete type 'class LinkedList<T>'

  176 | int LinkedList<T>::size(){

      |                         ^

In file included from C:\Users\guita\OneDrive\Documents\Arduino\libraries\ESPAsyncWebServer-master\src/ESPAsyncWebServer.h:29,

                 from Z:\Projects\3D_Printer\Split_Flap\split-flap-master\ESPMaster\ESPMaster.ino:10:

C:\Users\guita\OneDrive\Documents\Arduino\libraries\ESPAsyncWebServer-master\src/StringArray.h:39:7: note: declaration of 'class LinkedList<T>'

   39 | class LinkedList {

      |       ^~~~~~~~~~

In file included from Z:\Projects\3D_Printer\Split_Flap\split-flap-master\ESPMaster\ESPMaster.ino:15:

C:\Users\guita\OneDrive\Documents\Arduino\libraries\LinkedList/LinkedList.h:181:46: error: invalid use of incomplete type 'class LinkedList<T>'

  181 | LinkedList<T>::LinkedList(int sizeIndex, T _t){

      |                                              ^

In file included from C:\Users\guita\OneDrive\Documents\Arduino\libraries\ESPAsyncWebServer-master\src/ESPAsyncWebServer.h:29,

                 from Z:\Projects\3D_Printer\Split_Flap\split-flap-master\ESPMaster\ESPMaster.ino:10:

C:\Users\guita\OneDrive\Documents\Arduino\libraries\ESPAsyncWebServer-master\src/StringArray.h:39:7: note: declaration of 'class LinkedList<T>'

   39 | class LinkedList {

      |       ^~~~~~~~~~

In file included from Z:\Projects\3D_Printer\Split_Flap\split-flap-master\ESPMaster\ESPMaster.ino:15:

C:\Users\guita\OneDrive\Documents\Arduino\libraries\LinkedList/LinkedList.h:188:40: error: invalid use of incomplete type 'class LinkedList<T>'

  188 | bool LinkedList<T>::add(int index, T _t){

      |                                        ^

In file included from C:\Users\guita\OneDrive\Documents\Arduino\libraries\ESPAsyncWebServer-master\src/ESPAsyncWebServer.h:29,

                 from Z:\Projects\3D_Printer\Split_Flap\split-flap-master\ESPMaster\ESPMaster.ino:10:

C:\Users\guita\OneDrive\Documents\Arduino\libraries\ESPAsyncWebServer-master\src/StringArray.h:39:7: note: declaration of 'class LinkedList<T>'

   39 | class LinkedList {

      |       ^~~~~~~~~~

In file included from Z:\Projects\3D_Printer\Split_Flap\split-flap-master\ESPMaster\ESPMaster.ino:15:

C:\Users\guita\OneDrive\Documents\Arduino\libraries\LinkedList/LinkedList.h:209:29: error: invalid use of incomplete type 'class LinkedList<T>'

  209 | bool LinkedList<T>::add(T _t){

      |                             ^

In file included from C:\Users\guita\OneDrive\Documents\Arduino\libraries\ESPAsyncWebServer-master\src/ESPAsyncWebServer.h:29,

                 from Z:\Projects\3D_Printer\Split_Flap\split-flap-master\ESPMaster\ESPMaster.ino:10:

C:\Users\guita\OneDrive\Documents\Arduino\libraries\ESPAsyncWebServer-master\src/StringArray.h:39:7: note: declaration of 'class LinkedList<T>'

   39 | class LinkedList {

      |       ^~~~~~~~~~

In file included from Z:\Projects\3D_Printer\Split_Flap\split-flap-master\ESPMaster\ESPMaster.ino:15:

C:\Users\guita\OneDrive\Documents\Arduino\libraries\LinkedList/LinkedList.h:232:33: error: invalid use of incomplete type 'class LinkedList<T>'

  232 | bool LinkedList<T>::unshift(T _t){

      |                                 ^

In file included from C:\Users\guita\OneDrive\Documents\Arduino\libraries\ESPAsyncWebServer-master\src/ESPAsyncWebServer.h:29,

                 from Z:\Projects\3D_Printer\Split_Flap\split-flap-master\ESPMaster\ESPMaster.ino:10:

C:\Users\guita\OneDrive\Documents\Arduino\libraries\ESPAsyncWebServer-master\src/StringArray.h:39:7: note: declaration of 'class LinkedList<T>'

   39 | class LinkedList {

      |       ^~~~~~~~~~

In file included from Z:\Projects\3D_Printer\Split_Flap\split-flap-master\ESPMaster\ESPMaster.ino:15:

C:\Users\guita\OneDrive\Documents\Arduino\libraries\LinkedList/LinkedList.h:250:39: error: invalid use of incomplete type 'class LinkedList<T>'

  250 | T& LinkedList<T>::operator[](int index) {

      |                                       ^

In file included from C:\Users\guita\OneDrive\Documents\Arduino\libraries\ESPAsyncWebServer-master\src/ESPAsyncWebServer.h:29,

                 from Z:\Projects\3D_Printer\Split_Flap\split-flap-master\ESPMaster\ESPMaster.ino:10:

C:\Users\guita\OneDrive\Documents\Arduino\libraries\ESPAsyncWebServer-master\src/StringArray.h:39:7: note: declaration of 'class LinkedList<T>'

   39 | class LinkedList {

      |       ^~~~~~~~~~

In file included from Z:\Projects\3D_Printer\Split_Flap\split-flap-master\ESPMaster\ESPMaster.ino:15:

C:\Users\guita\OneDrive\Documents\Arduino\libraries\LinkedList/LinkedList.h:255:40: error: invalid use of incomplete type 'class LinkedList<T>'

  255 | bool LinkedList<T>::set(int index, T _t){

      |                                        ^

In file included from C:\Users\guita\OneDrive\Documents\Arduino\libraries\ESPAsyncWebServer-master\src/ESPAsyncWebServer.h:29,

                 from Z:\Projects\3D_Printer\Split_Flap\split-flap-master\ESPMaster\ESPMaster.ino:10:

C:\Users\guita\OneDrive\Documents\Arduino\libraries\ESPAsyncWebServer-master\src/StringArray.h:39:7: note: declaration of 'class LinkedList<T>'

   39 | class LinkedList {

      |       ^~~~~~~~~~

In file included from Z:\Projects\3D_Printer\Split_Flap\split-flap-master\ESPMaster\ESPMaster.ino:15:

C:\Users\guita\OneDrive\Documents\Arduino\libraries\LinkedList/LinkedList.h:265:22: error: invalid use of incomplete type 'class LinkedList<T>'

  265 | T LinkedList<T>::pop(){

      |                      ^

In file included from C:\Users\guita\OneDrive\Documents\Arduino\libraries\ESPAsyncWebServer-master\src/ESPAsyncWebServer.h:29,

                 from Z:\Projects\3D_Printer\Split_Flap\split-flap-master\ESPMaster\ESPMaster.ino:10:

C:\Users\guita\OneDrive\Documents\Arduino\libraries\ESPAsyncWebServer-master\src/StringArray.h:39:7: note: declaration of 'class LinkedList<T>'

   39 | class LinkedList {

      |       ^~~~~~~~~~

In file included from Z:\Projects\3D_Printer\Split_Flap\split-flap-master\ESPMaster\ESPMaster.ino:15:

C:\Users\guita\OneDrive\Documents\Arduino\libraries\LinkedList/LinkedList.h:291:24: error: invalid use of incomplete type 'class LinkedList<T>'

  291 | T LinkedList<T>::shift(){

      |                        ^

In file included from C:\Users\guita\OneDrive\Documents\Arduino\libraries\ESPAsyncWebServer-master\src/ESPAsyncWebServer.h:29,

                 from Z:\Projects\3D_Printer\Split_Flap\split-flap-master\ESPMaster\ESPMaster.ino:10:

C:\Users\guita\OneDrive\Documents\Arduino\libraries\ESPAsyncWebServer-master\src/StringArray.h:39:7: note: declaration of 'class LinkedList<T>'

   39 | class LinkedList {

      |       ^~~~~~~~~~

In file included from Z:\Projects\3D_Printer\Split_Flap\split-flap-master\ESPMaster\ESPMaster.ino:15:

C:\Users\guita\OneDrive\Documents\Arduino\libraries\LinkedList/LinkedList.h:312:34: error: invalid use of incomplete type 'class LinkedList<T>'

  312 | T LinkedList<T>::remove(int index){

      |                                  ^

In file included from C:\Users\guita\OneDrive\Documents\Arduino\libraries\ESPAsyncWebServer-master\src/ESPAsyncWebServer.h:29,

                 from Z:\Projects\3D_Printer\Split_Flap\split-flap-master\ESPMaster\ESPMaster.ino:10:

C:\Users\guita\OneDrive\Documents\Arduino\libraries\ESPAsyncWebServer-master\src/StringArray.h:39:7: note: declaration of 'class LinkedList<T>'

   39 | class LinkedList {

      |       ^~~~~~~~~~

In file included from Z:\Projects\3D_Printer\Split_Flap\split-flap-master\ESPMaster\ESPMaster.ino:15:

C:\Users\guita\OneDrive\Documents\Arduino\libraries\LinkedList/LinkedList.h:338:31: error: invalid use of incomplete type 'class LinkedList<T>'

  338 | T LinkedList<T>::get(int index){

      |                               ^

In file included from C:\Users\guita\OneDrive\Documents\Arduino\libraries\ESPAsyncWebServer-master\src/ESPAsyncWebServer.h:29,

                 from Z:\Projects\3D_Printer\Split_Flap\split-flap-master\ESPMaster\ESPMaster.ino:10:

C:\Users\guita\OneDrive\Documents\Arduino\libraries\ESPAsyncWebServer-master\src/StringArray.h:39:7: note: declaration of 'class LinkedList<T>'

   39 | class LinkedList {

      |       ^~~~~~~~~~

In file included from Z:\Projects\3D_Printer\Split_Flap\split-flap-master\ESPMaster\ESPMaster.ino:15:

C:\Users\guita\OneDrive\Documents\Arduino\libraries\LinkedList/LinkedList.h:345:27: error: invalid use of incomplete type 'class LinkedList<T>'

  345 | void LinkedList<T>::clear(){

      |                           ^

In file included from C:\Users\guita\OneDrive\Documents\Arduino\libraries\ESPAsyncWebServer-master\src/ESPAsyncWebServer.h:29,

                 from Z:\Projects\3D_Printer\Split_Flap\split-flap-master\ESPMaster\ESPMaster.ino:10:

C:\Users\guita\OneDrive\Documents\Arduino\libraries\ESPAsyncWebServer-master\src/StringArray.h:39:7: note: declaration of 'class LinkedList<T>'

   39 | class LinkedList {

      |       ^~~~~~~~~~

In file included from Z:\Projects\3D_Printer\Split_Flap\split-flap-master\ESPMaster\ESPMaster.ino:15:

C:\Users\guita\OneDrive\Documents\Arduino\libraries\LinkedList/LinkedList.h:351:46: error: invalid use of incomplete type 'class LinkedList<T>'

  351 | void LinkedList<T>::sort(int (*cmp)(T &, T &)){

      |                                              ^

In file included from C:\Users\guita\OneDrive\Documents\Arduino\libraries\ESPAsyncWebServer-master\src/ESPAsyncWebServer.h:29,

                 from Z:\Projects\3D_Printer\Split_Flap\split-flap-master\ESPMaster\ESPMaster.ino:10:

C:\Users\guita\OneDrive\Documents\Arduino\libraries\ESPAsyncWebServer-master\src/StringArray.h:39:7: note: declaration of 'class LinkedList<T>'

   39 | class LinkedList {

      |       ^~~~~~~~~~

In file included from Z:\Projects\3D_Printer\Split_Flap\split-flap-master\ESPMaster\ESPMaster.ino:15:

C:\Users\guita\OneDrive\Documents\Arduino\libraries\LinkedList/LinkedList.h:411:87: error: invalid use of incomplete type 'class LinkedList<T>'

  411 | ListNode<T>* LinkedList<T>::findEndOfSortedString(ListNode<T> *p, int (*cmp)(T &, T &)) {

      |                                                                                       ^

In file included from C:\Users\guita\OneDrive\Documents\Arduino\libraries\ESPAsyncWebServer-master\src/ESPAsyncWebServer.h:29,

                 from Z:\Projects\3D_Printer\Split_Flap\split-flap-master\ESPMaster\ESPMaster.ino:10:

C:\Users\guita\OneDrive\Documents\Arduino\libraries\ESPAsyncWebServer-master\src/StringArray.h:39:7: note: declaration of 'class LinkedList<T>'

   39 | class LinkedList {

      |       ^~~~~~~~~~

ESPMaster:55:30: error: no matching function for call to 'LinkedList<ScheduledMessage>::LinkedList()'

   55 | LinkedList<ScheduledMessage> scheduledMessages;

      |                              ^~~~~~~~~~~~~~~~~

In file included from C:\Users\guita\OneDrive\Documents\Arduino\libraries\ESPAsyncWebServer-master\src/ESPAsyncWebServer.h:29,

                 from Z:\Projects\3D_Printer\Split_Flap\split-flap-master\ESPMaster\ESPMaster.ino:10:

C:\Users\guita\OneDrive\Documents\Arduino\libraries\ESPAsyncWebServer-master\src/StringArray.h:64:5: note: candidate: 'LinkedList<T, Item>::LinkedList(LinkedList<T, Item>::OnRemove) [with T = ScheduledMessage; Item = LinkedListNode; LinkedList<T, Item>::OnRemove = std::function<void(const ScheduledMessage&)>]'

   64 |     LinkedList(OnRemove onRemove) : _root(nullptr), _onRemove(onRemove) {}

      |     ^~~~~~~~~~

C:\Users\guita\OneDrive\Documents\Arduino\libraries\ESPAsyncWebServer-master\src/StringArray.h:64:5: note:   candidate expects 1 argument, 0 provided

C:\Users\guita\OneDrive\Documents\Arduino\libraries\ESPAsyncWebServer-master\src/StringArray.h:39:7: note: candidate: 'LinkedList<ScheduledMessage>::LinkedList(const LinkedList<ScheduledMessage>&)'

   39 | class LinkedList {

      |       ^~~~~~~~~~

C:\Users\guita\OneDrive\Documents\Arduino\libraries\ESPAsyncWebServer-master\src/StringArray.h:39:7: note:   candidate expects 1 argument, 0 provided

Z:\Projects\3D_Printer\Split_Flap\split-flap-master\ESPMaster\ESPMaster.ino: In lambda function:

ESPMaster:255:84: error: 'class LinkedList<ScheduledMessage>' has no member named 'size'

  255 |       for(int scheduledMessageIndex = 0; scheduledMessageIndex < scheduledMessages.size(); scheduledMessageIndex++) {

      |                                                                                    ^~~~

ESPMaster:256:62: error: no match for 'operator[]' (operand types are 'LinkedList<ScheduledMessage>' and 'int')

  256 |         ScheduledMessage scheduledMessage = scheduledMessages[scheduledMessageIndex];

      |                                                              ^

ESPMaster:260:36: error: cannot convert 'int' to 'const ScheduledMessage&'

  260 |           scheduledMessages.remove(scheduledMessageIndex);

      |                                    ^~~~~~~~~~~~~~~~~~~~~

      |                                    |

      |                                    int

In file included from C:\Users\guita\OneDrive\Documents\Arduino\libraries\ESPAsyncWebServer-master\src/ESPAsyncWebServer.h:29,

                 from Z:\Projects\3D_Printer\Split_Flap\split-flap-master\ESPMaster\ESPMaster.ino:10:

C:\Users\guita\OneDrive\Documents\Arduino\libraries\ESPAsyncWebServer-master\src/StringArray.h:116:26: note:   initializing argument 1 of 'bool LinkedList<T, Item>::remove(const T&) [with T = ScheduledMessage; Item = LinkedListNode]'

  116 |     bool remove(const T& t){

      |                 ~~~~~~~~~^

Z:\Projects\3D_Printer\Split_Flap\split-flap-master\ESPMaster\ESPMaster.ino: In function 'void checkScheduledMessages()':

ESPMaster:425:82: error: 'class LinkedList<ScheduledMessage>' has no member named 'size'

  425 |     for(int scheduledMessageIndex = 0; scheduledMessageIndex < scheduledMessages.size(); scheduledMessageIndex++) {

      |                                                                                  ^~~~

ESPMaster:426:60: error: no match for 'operator[]' (operand types are 'LinkedList<ScheduledMessage>' and 'int')

  426 |       ScheduledMessage scheduledMessage = scheduledMessages[scheduledMessageIndex];

      |                                                            ^

ESPMaster:430:34: error: cannot convert 'int' to 'const ScheduledMessage&'

  430 |         scheduledMessages.remove(scheduledMessageIndex);

      |                                  ^~~~~~~~~~~~~~~~~~~~~

      |                                  |

      |                                  int

In file included from C:\Users\guita\OneDrive\Documents\Arduino\libraries\ESPAsyncWebServer-master\src/ESPAsyncWebServer.h:29,

                 from Z:\Projects\3D_Printer\Split_Flap\split-flap-master\ESPMaster\ESPMaster.ino:10:

C:\Users\guita\OneDrive\Documents\Arduino\libraries\ESPAsyncWebServer-master\src/StringArray.h:116:26: note:   initializing argument 1 of 'bool LinkedList<T, Item>::remove(const T&) [with T = ScheduledMessage; Item = LinkedListNode]'

  116 |     bool remove(const T& t){

      |                 ~~~~~~~~~^

ESPMaster:454:80: error: 'class LinkedList<ScheduledMessage>' has no member named 'size'

  454 |   for(int scheduledMessageIndex = 0; scheduledMessageIndex < scheduledMessages.size(); scheduledMessageIndex++) {

      |                                                                                ^~~~

ESPMaster:455:58: error: no match for 'operator[]' (operand types are 'LinkedList<ScheduledMessage>' and 'int')

  455 |     ScheduledMessage scheduledMessage = scheduledMessages[scheduledMessageIndex];

      |                                                          ^

ESPMaster:467:32: error: cannot convert 'int' to 'const ScheduledMessage&'

  467 |       scheduledMessages.remove(scheduledMessageIndex);

      |                                ^~~~~~~~~~~~~~~~~~~~~

      |                                |

      |                                int

In file included from C:\Users\guita\OneDrive\Documents\Arduino\libraries\ESPAsyncWebServer-master\src/ESPAsyncWebServer.h:29,

                 from Z:\Projects\3D_Printer\Split_Flap\split-flap-master\ESPMaster\ESPMaster.ino:10:

C:\Users\guita\OneDrive\Documents\Arduino\libraries\ESPAsyncWebServer-master\src/StringArray.h:116:26: note:   initializing argument 1 of 'bool LinkedList<T, Item>::remove(const T&) [with T = ScheduledMessage; Item = LinkedListNode]'

  116 |     bool remove(const T& t){

      |                 ~~~~~~~~~^

Z:\Projects\3D_Printer\Split_Flap\split-flap-master\ESPMaster\HelpersStringHandling.ino: In function 'LinkedList<String> processSentenceToLines(String)':

HelpersStringHandling:91:22: error: no matching function for call to 'LinkedList<String>::LinkedList()'

   91 |   LinkedList<String> words;

      |                      ^~~~~

In file included from C:\Users\guita\OneDrive\Documents\Arduino\libraries\ESPAsyncWebServer-master\src/ESPAsyncWebServer.h:29,

                 from Z:\Projects\3D_Printer\Split_Flap\split-flap-master\ESPMaster\ESPMaster.ino:10:

C:\Users\guita\OneDrive\Documents\Arduino\libraries\ESPAsyncWebServer-master\src/StringArray.h:64:5: note: candidate: 'LinkedList<T, Item>::LinkedList(LinkedList<T, Item>::OnRemove) [with T = String; Item = LinkedListNode; LinkedList<T, Item>::OnRemove = std::function<void(const String&)>]'

   64 |     LinkedList(OnRemove onRemove) : _root(nullptr), _onRemove(onRemove) {}

      |     ^~~~~~~~~~

C:\Users\guita\OneDrive\Documents\Arduino\libraries\ESPAsyncWebServer-master\src/StringArray.h:64:5: note:   candidate expects 1 argument, 0 provided

C:\Users\guita\OneDrive\Documents\Arduino\libraries\ESPAsyncWebServer-master\src/StringArray.h:39:7: note: candidate: 'LinkedList<String>::LinkedList(const LinkedList<String>&)'

   39 | class LinkedList {

      |       ^~~~~~~~~~

C:\Users\guita\OneDrive\Documents\Arduino\libraries\ESPAsyncWebServer-master\src/StringArray.h:39:7: note:   candidate expects 1 argument, 0 provided

HelpersStringHandling:112:22: error: no matching function for call to 'LinkedList<String>::LinkedList()'

  112 |   LinkedList<String> lines;

      |                      ^~~~~

In file included from C:\Users\guita\OneDrive\Documents\Arduino\libraries\ESPAsyncWebServer-master\src/ESPAsyncWebServer.h:29,

                 from Z:\Projects\3D_Printer\Split_Flap\split-flap-master\ESPMaster\ESPMaster.ino:10:

C:\Users\guita\OneDrive\Documents\Arduino\libraries\ESPAsyncWebServer-master\src/StringArray.h:64:5: note: candidate: 'LinkedList<T, Item>::LinkedList(LinkedList<T, Item>::OnRemove) [with T = String; Item = LinkedListNode; LinkedList<T, Item>::OnRemove = std::function<void(const String&)>]'

   64 |     LinkedList(OnRemove onRemove) : _root(nullptr), _onRemove(onRemove) {}

      |     ^~~~~~~~~~

C:\Users\guita\OneDrive\Documents\Arduino\libraries\ESPAsyncWebServer-master\src/StringArray.h:64:5: note:   candidate expects 1 argument, 0 provided

C:\Users\guita\OneDrive\Documents\Arduino\libraries\ESPAsyncWebServer-master\src/StringArray.h:39:7: note: candidate: 'LinkedList<String>::LinkedList(const LinkedList<String>&)'

   39 | class LinkedList {

      |       ^~~~~~~~~~

C:\Users\guita\OneDrive\Documents\Arduino\libraries\ESPAsyncWebServer-master\src/StringArray.h:39:7: note:   candidate expects 1 argument, 0 provided

HelpersStringHandling:114:15: error: 'class LinkedList<String>' has no member named 'size'

  114 |   while(words.size() != 0)

      |               ^~~~

HelpersStringHandling:117:28: error: no match for 'operator[]' (operand types are 'LinkedList<String>' and 'int')

  117 |     String wordItem = words[0];

      |                            ^

HelpersStringHandling:174:18: error: conversion from 'int' to 'const String' is ambiguous

  174 |     words.remove(0);

      |                  ^

In file included from C:\Users\guita\AppData\Local\Arduino15\packages\esp8266\hardware\esp8266\3.1.2\cores\esp8266/Arduino.h:295,

                 from Z:\Projects\3D_Printer\Split_Flap\split-flap-master\ESPMaster\ESPMaster.ino:6:

C:\Users\guita\AppData\Local\Arduino15\packages\esp8266\hardware\esp8266\3.1.2\cores\esp8266/WString.h:59:9: note: candidate: 'String::String(const __FlashStringHelper*)'

   59 |         String(const __FlashStringHelper *str);

      |         ^~~~~~

C:\Users\guita\AppData\Local\Arduino15\packages\esp8266\hardware\esp8266\3.1.2\cores\esp8266/WString.h:57:9: note: candidate: 'String::String(const char*)'

   57 |         String(const char *cstr);

      |         ^~~~~~

In file included from C:\Users\guita\OneDrive\Documents\Arduino\libraries\ESPAsyncWebServer-master\src/ESPAsyncWebServer.h:29,

                 from Z:\Projects\3D_Printer\Split_Flap\split-flap-master\ESPMaster\ESPMaster.ino:10:

C:\Users\guita\OneDrive\Documents\Arduino\libraries\ESPAsyncWebServer-master\src/StringArray.h:116:26: note:   initializing argument 1 of 'bool LinkedList<T, Item>::remove(const T&) [with T = String; Item = LinkedListNode]'

  116 |     bool remove(const T& t){

      |                 ~~~~~~~~~^

HelpersStringHandling:177:15: error: 'class LinkedList<String>' has no member named 'size'

  177 |     if (words.size() == 0 && inProgressLine != "")

      |               ^~~~

Z:\Projects\3D_Printer\Split_Flap\split-flap-master\ESPMaster\ServiceFileSystemFunctions.ino: In function 'String getCurrentSettingValues()':

ServiceFileSystemFunctions:35:80: error: 'class LinkedList<ScheduledMessage>' has no member named 'size'

   35 |   for(int scheduledMessageIndex = 0; scheduledMessageIndex < scheduledMessages.size(); scheduledMessageIndex++) {

      |                                                                                ^~~~

ServiceFileSystemFunctions:36:58: error: no match for 'operator[]' (operand types are 'LinkedList<ScheduledMessage>' and 'int')

   36 |     ScheduledMessage scheduledMessage = scheduledMessages[scheduledMessageIndex];

      |                                                          ^

Z:\Projects\3D_Printer\Split_Flap\split-flap-master\ESPMaster\ServiceFlapFunctions.ino: In function 'void showText(String)':

ServiceFlapFunctions:10:22: error: 'class LinkedList<String>' has no member named 'size'

   10 |     if (messageLines.size() > 1) {

      |                      ^~~~

ServiceFlapFunctions:14:58: error: 'class LinkedList<String>' has no member named 'size'

   14 |       for (int linesIndex = 0; linesIndex < messageLines.size(); linesIndex++) {

      |                                                          ^~~~

ServiceFlapFunctions:15:35: error: no match for 'operator[]' (operand types are 'LinkedList<String>' and 'int')

   15 |         String line = messageLines[linesIndex];

      |                                   ^

ServiceFlapFunctions:23:40: error: 'class LinkedList<String>' has no member named 'size'

   23 |         if (linesIndex <= messageLines.size()) {

      |                                        ^~~~

exit status 1

no matching function for call to 'LinkedList<ScheduledMessage>::LinkedList()'

This report would have more information with
"Show verbose output during compilation"
option enabled in File -> Preferences.

**Comment by chris-guthrie** [comment]
The problem is related to LinkedList library clashing with ESPAsyncWebServer, you can see a work around listed on this issue report.

https://github.com/ivanseidel/LinkedList/issues/40

**Comment by Lo0kadistraction** [comment]
That's a good fix include the LList.h and refactor all LinkedList references to LList and its good to go. thank you!

**Comment by Lo0kadistraction** [comment]
That's a good fix include the LList.h and refactor all LinkedList references to LList and its good to go. thank you!

**Comment by JonnyBooker** [comment]
@Lo0kadistraction I am so so sorry! I never saw this get opened as an issue! I will try pay more close attention. Hope the rest of the build went more smoothly for you

@stokes776 I'm glad you solved your issue. The way I got around it was by modifying the original ESPAsyncWebServer library and including it as a zip here in this repository. The wording I used for including the libraries I can look to improve in the readme.
